### PR TITLE
Fix/httplog generalized the plugin for re-use

### PR DESCRIPTION
**Pull request by Tieske**
includes #1689 and adds some comment updates and exposing the conf table to the serializer (in case a different content-type is configured)

### CODE REVIEW

File: kong/plugins/http-log/handler.lua [hunk 1]
```diff
@@ -39,9 +47,11 @@ local function parse_url(host_url)
 end
 
 -- Log to a Http end point.
--- @param `premature`
--- @param `conf`     Configuration table, holds http endpoint details
--- @param `message`  Message to be logged
+-- This basically is structured as a timer callback.
+-- @param `premature` see openresty timer_at function
```
Comment (WALL-E) [inline kong/plugins/http-log/handler.lua]: @param `premature` see openresty ngx.timer.at function

Comment (Tieske) [inline kong/plugins/http-log/handler.lua]: thx. fixed

**Comment by Tieske** [comment]
@WALL-E merged the generic modifications, you can now update the influxdb one based of this.

**Comment by WALL-E** [comment]
@Tieske ok, I will fix it

**Pull request merged**